src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Receivers.kt

```diff
@@ -181,36 +190,37 @@ class MarkdownTaskReceiver(
 
     private fun getCmdLineArguments(request: CompileRequest): Map<String, String> {
         val outputFileName = Files.getNameWithoutExtension(request.mainFileName) + ".tex"
-        val projTasks = this.tasksDir.resolve(request.id)
-        val mainFile = request.mainFileName
-        val outputFile = projTasks.resolve(outputFileName)
-        val projectRootAbsPath = this.tasksDir.toAbsolutePath().parent
+        val projTasks = this.tasksDir.resolve(request.id).resolve("files")
+        val mainFile = StringUtils.stripStart(request.mainFileName, "/")
+        val values = listOf(projTasks.toString(), "",
+                mainFile, outputFileName, PANDOC_DEFAULT_FONT)
 
-        val values = listOf(projectRootAbsPath.toString(), "",
-                mainFile.toString(), outputFile.toString(), PANDOC_DEFAULT_FONT)
-
-        return (arguments zip values).map { it.first to it.second }.toMap()
+        return (arguments zip values).map { it.first to "\"${it.second}\"" }.toMap()
     }
 
     // converts Markdown into tex via pandoc
     private fun convertMarkdown(commandArguments: Map<String, String>): Int {
         val substitutor = StringSubstitutor(commandArguments)
         val rawCommandLine = config.getString(COMPILE_COMMAND_KEY)
         val commandLine = substitutor.replace(rawCommandLine)
-        return runCommandLine(commandLine)
+        return runCommandLine(commandLine.replace("\\", ""))
```

Comment:
??? Why would anyone wrote escaped slash in the command line in he knew it would have been removed by this piece of code?

Reply:
Yeah, it's pretty weird. I just wanted to removed leading backslash after the config string, because otherwise, due to quotes, arguments will be like \"arg_content"